### PR TITLE
Fix missing legend in Bar chart 

### LINKED ISSUE #111: legendDataForKeys() missing data
``` javascript
 var legendDataForKeys = result.bars.filter(function (bar) {
        return bar.data.index === 0;
    }).map(function (bar) {
        return {
            label: bar.data.id,
            fill: bar.data.fill ? bar.data.fill : bar.color
        };
    }).reverse();
```

Consider following data:

```json
[
  {
    key1: 0,
    key2: 100,
    key3: 100,
    idx: 0
  },
  {
    key1:100,
    key2:300,
    key3:400,
    idx:1
  }
]
```

Key1 would be missed in legendDataForKeys, as it seems that result.bars only contains those that have visible heights?

### PR DESCRIPTION
**Pull request by AhyoungRyu**
Another trial to fix #111 or #218 

So the problem that I aim to fix here is, if some keys have `0` as value in first column, it doesn't generate legend at all even if other columns have non-zero value. 

Related conversation is also in #128 which is a first trial to fix this issue. I saw #219 as well and I understand there might be more cases we can improve. 
But I think many ppl including me, this missing legend with zero value in 1st column issue is more urgent I think. 

tl; dr 

#### Before 
screen shot 2018-11-30 at 2 17 13 pm

#### After 
screen shot 2018-11-30 at 2 17 04 pm

Also applied in Canvas 
screen shot 2018-11-30 at 2 12 51 pm

### CODE REVIEW

**Comment by justincrow** [comment]
awesome - when will this fix be published?

**Comment by AhyoungRyu** [comment]
Yeah I also wondered too ;) 
@plouc When will be the next release ?

**Comment by stahlmanDesign** [comment]
I saw this was finally fixed with v0.52.0!
capture d ecran 2018-12-10 a 17 37 06

**Comment by AntonPuko** [comment]
ehhh..can this be evaluated not only  in the first bar?
with current calculations its not possible to use legend in realtime chart like this:
image
where the first bar can be empty.
:(
